java/demo/src/com/google/phonenumbers/PhoneNumberParserServlet.java

```diff
@@ -214,6 +217,29 @@ private StringBuilder getOutputForSingleNumber(
       output.append("</TABLE>");
       output.append("</DIV>");
 
+      boolean isPossibleShort = shortInfo.isPossibleShortNumber(number);
```

Comment:
Do we want to do this every time, or only if it isn't a valid number at all? Will we/should we have numbers that are both? Too much output can be confusing. You could instead of this just above have a bit where if it's invalid, put this possible/valid short number info.